integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/processor/test/ESBJAVA4819DynamicallySettingWSAHeadersTestCase.java

```diff
@@ -0,0 +1,67 @@
+/*
+*Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*WSO2 Inc. licenses this file to you under the Apache License,
+*Version 2.0 (the "License"); you may not use this file except
+*in compliance with the License.
+*You may obtain a copy of the License at
+*
+*http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing,
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+package org.wso2.carbon.esb.message.processor.test;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.test.utils.http.client.HttpURLConnectionClient;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.Utils;
+
+import java.io.Reader;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.net.URL;
+
+/**
+ * Test case to test whether message processor is able to handle messages when their WSA headers are set dynamically.
+ * Test case for fix ESBJAVA-4819.
+ */
+public class ESBJAVA4819DynamicallySettingWSAHeadersTestCase extends ESBIntegrationTest {
```

Comment:
We name the test case as ESBJAVA4819... only in the support branch. For the public branch, we need to have a sensible name describing the scenario that is being tested. Shall we correct?

Reply:
done